src/plugins/data/public/search/search_interceptor.ts

```diff
@@ -84,6 +86,41 @@ export class SearchInterceptor {
     });
   }
 
+  /*
+   * @returns `TimeoutErrorMode` indicating what action should be taken in case of a request timeout based on license and permissions.
+   * @internal
+   */
+  protected getTimeoutMode() {
+    return TimeoutErrorMode.UPGRADE;
+  }
+
+  /*
+   * @returns `Error` a search service specific error or the original error, if a specific error can't be recognized.
+   * @internal
+   */
+  protected getSearchError(
+    e: any,
+    request: IEsSearchRequest,
+    timeoutSignal: AbortSignal,
+    appAbortSignal?: AbortSignal
+  ): Error {
+    if (timeoutSignal.aborted || get(e, 'body.message') === 'Request timed out') {
+      // Handle a client or a server side timeout
+      const err = new SearchTimeoutError(e, this.getTimeoutMode());
+
+      // Show the timeout error here, so that it's shown regardless of how an application chooses to handle errors.
+      this.showTimeoutError(err);
```

Comment:
I think this better be moved outside this function (or function remanded into `handleSearchError`)